pkg/shortener/shortener.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package shortener
+
+import (
+	"crypto/sha256"
+	"fmt"
+)
+
+const (
+	// Maximum characters in a K8s resource name
+	k8sMaxResourceNameLength = 63
+)
+
+func ShortenK8sResourceName(s string) string {
```

Comment:
I don't want to block the PR on it, but could use a bit of unit testing to document/encode that a property this needs to have is that it's pure/stable in the sense that it must return the same short name for a given input.

Reply:
Yep, makes sense. I'll add that.

Reply:
Added